include/rapidjson/reader.h

```diff
@@ -462,6 +463,29 @@ class GenericReader {
         ClearStackOnExit& operator=(const ClearStackOnExit&);
     };
 
+    template<unsigned parseFlags, typename InputStream>
+    void SkipWhitespaceAndComments(InputStream& is) {
+        SkipWhitespace(is);
+
+        if (parseFlags & kParseCommentsFlag) {
+            while (is.Peek() == '/') {
+                is.Take();
+
+                if (is.Peek() == '*') {
+                    is.Take();
+                    while (is.Take() != '*' || is.Take() != '/') { }

```

Comment:
It seems need to check for null terminator here. Otherwise a malformed input with only open `/*` may read after null terminator.